v2/internal/attacktechniques/azure/persistence/bastion-shareable-link/main.tf

```diff
@@ -0,0 +1,171 @@
+  location            = azurerm_resource_group.lab_environment.location
+  resource_group_name = azurerm_resource_group.lab_environment.name
+}
+
+resource "azurerm_subnet" "bastion_subnet" {
+  # Required naming for deployment of Azure Bastion
+  name                 = "AzureBastionSubnet"
+  resource_group_name  = azurerm_resource_group.lab_environment.name
+  virtual_network_name = azurerm_virtual_network.lab_vnet.name
+  address_prefixes     = ["10.0.0.0/27"]
+}
+
+resource "azurerm_subnet" "lab_subnet" {
+  name                 = "${local.resource_prefix}-subnet-${random_string.lab_name.result}"
+  resource_group_name  = azurerm_resource_group.lab_environment.name
+  virtual_network_name = azurerm_virtual_network.lab_vnet.name
+  address_prefixes     = ["10.0.0.32/27"]
+}
+
+resource "azurerm_public_ip" "lab_pip" {
+  name                = "${local.resource_prefix}-pip-${random_string.lab_name.result}"
+  location            = azurerm_resource_group.lab_environment.location
+  resource_group_name = azurerm_resource_group.lab_environment.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+
+resource "azurerm_network_interface" "lab_nic" {
+  name                = "${local.resource_prefix}-nic-${random_string.lab_name.result}"
+  location            = azurerm_resource_group.lab_environment.location
+  resource_group_name = azurerm_resource_group.lab_environment.name
+
+  ip_configuration {
+    name                          = "${local.resource_prefix}-ip-${random_string.lab_name.result}"
+    subnet_id                     = azurerm_subnet.lab_subnet.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# Virtual Machine Resources
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+resource "azurerm_windows_virtual_machine" "lab_windows_vm" {
+  name                = "srt-vm-bsl" # 15 character limit: stratus red team - vm - bastion shareable link
+  resource_group_name = azurerm_resource_group.lab_environment.name
+  location            = azurerm_resource_group.lab_environment.location
+  size                = "Standard_F2"
+  admin_username      = "local_admin_user"
+  admin_password      = random_password.password.result
+  user_data           = base64encode(random_string.lab_name.result)
+
+  network_interface_ids = [
+    azurerm_network_interface.lab_nic.id,
+  ]
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2016-Datacenter"
+    version   = "latest"
+  }
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# Bastion Resource
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+# Note: Creation/destruction of a Bastion can take 10 minutes each, see https://learn.microsoft.com/en-us/azure/bastion/tutorial-create-host-portal
```

Comment:
oh dear

Reply:
Yeah, I wish there were a way to speed this up.